src/app/static/src/app/apiService.ts

```diff
@@ -120,7 +120,12 @@ export class APIService<S extends string, E extends string> implements API<S, E>
             return
         }
 
-        let failure = e.response && e.response.data;
+        if (e.response && e.response.status == 401) {
+            window.location.assign("/login")
```

Comment:
I think we also said that we wanted the login page to show a message to the user, something your 'Your session expired. Please login again to continue.'  I think you can do this by including the message you want to show in a query string parameter 'error' to the login endpoint. (As handled by the backend here: https://github.com/mrc-ide/hint/blob/master/src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/LoginController.kt)
This is nice, because at this point, we still know what the user's preferred language is, so we can keep the message in a locale string..

Reply:
Implemented